### PR TITLE
net: lib: aws_fota: Remove unused parameter

### PR DESCRIPTION
**Pull request by sigvartmh**
Remove `app_version` parameter as it's unused. It was used to update the device shadow inside the library. Updating device shadow has been moved out of the library, and into the sample. Therefore this is not needed anymore.

Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>

### CODE REVIEW

**Comment by sigvartmh** [comment]
Yup will update it after CI is green.

**Comment by sigvartmh** [comment]
Change log updated.

File: subsys/net/lib/aws_fota/src/aws_fota.c [hunk 1]
```diff
@@ -509,10 +508,6 @@ int aws_fota_init(struct mqtt_client *const client,
 		return -EINVAL;
 	}
 
-	if (strlen(app_version) >= CONFIG_AWS_FOTA_VERSION_STRING_MAX_LEN) {
```
Comment (oyvindronningstad) [inline subsys/net/lib/aws_fota/src/aws_fota.c]: This check is no longer needed?

Comment (sigvartmh) [reply subsys/net/lib/aws_fota/src/aws_fota.c]: `app_version` is removed so yes.

Comment (sigvartmh) [reply subsys/net/lib/aws_fota/src/aws_fota.c]: Bit seems CONFIG_AWS_FOTA_VERSION_STRING_MAX_LEN also should be removed I'll remove it.

Comment (sigvartmh) [reply subsys/net/lib/aws_fota/src/aws_fota.c]: It was removed my bad :) is removed from kconfig AWS_FOTA_VERSION_STRING_MAX_LEN

**Comment by sigvartmh** [comment]
Ready to merge @hakonfam @rlubos @lemrey

**Comment by sigvartmh** [comment]
@rlubos @lemrey  CI is green feel free to merge

**Pull request merged**